src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,96 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        threshold = (int) (INITIAL_CAPACITY * LOAD_FACTOR);
+        table = new Node[INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size == threshold) {
+            resize();
+        }
+        Node<K, V> newNode = new Node<>(key, value, null);
+        int index = findIndex(hash(key));
+        Node<K, V> iterator = table[index];
+        while (iterator != null) {
+            if (Objects.equals(iterator.key, key)) {
+                iterator.value = newNode.value;
+                return;
+            }
+            if (iterator.next == null) {
+                iterator.next = newNode;
+                size++;
+                return;
+            }
+            iterator = iterator.next;
+        }
+        table[index] = newNode;
```

Comment:
```suggestion
        iterator = newNode;
```
can we remove second `if` in `while` loop?